ocaml-lsp-server/src/folding_range.ml

```diff
@@ -16,6 +16,17 @@ let fold_over_parsetree (parsetree : Mreader.parsetree) =
         (typ_decl : Parsetree.type_declaration) =
       Range.of_loc typ_decl.ptype_loc |> push
     in
+    let type_extension (_self : Ast_iterator.iterator)
+        (typ_ext : Parsetree.type_extension) =
+      let loc = typ_ext.ptyext_path.loc in
+      let last_constr = List.last typ_ext.ptyext_constructors in
+      let loc =
+        match last_constr with
+        | None -> loc
+        | Some { pext_loc; _ } -> { loc with loc_end = pext_loc.loc_end }
+      in
+      Range.of_loc loc |> push
```

Comment:
I didn't check this code very closely, but it seems strange to me that we always push to this stack but never pop. could you explain why that's correct?

Reply:
`Folding_range.fold_over_parsetree` initializes a new stack for each request. The stack is `rev_map`ed and returned at the end of the function. Am I missing something?